### DISCUSSION TITLE
How to get only one user's commit info?

### DISCUSSION THREAD

**Discussion by ArivAfonso** [community/community#24667] [Apps, API and Webhooks]
<p>Even though github has a <a href="https://docs.github.com/en/rest/reference/repository-metrics#get-all-contributor-commit-activity" rel="noopener nofollow ugc">Repository Metrics Api</a> which lists all activity. It lists the commits of all contributors and for big projects, the response is often to big too handle. It would be really appreciated if there was an optional author parameter at the end such as</p>
<p>/repos/{owner}/{repo}/stats/contributors?author={name}</p>
<p>If it is not possible, is there a way to do this in grapql?</p>

**Comment by jfullstackdev** [comment]
<p>yes you can always do that by getting only the required field, like in JavaScript, right?</p>

**Comment by ArivAfonso** [comment]
<p>Could you please elaborate?</p>

**Comment by jfullstackdev** [comment, marked as answer]
<p>for example, in one of my private app, I consume one endpoint, and this is the code snippet,</p>
<pre><code class="lang-auto">await octokit.request('GET /user/following/{username}', {
    username: following.data[i].login
})
</code></pre>
<p>that <code>login</code> is the username of those I’m following, I loop that, so I can access each username, and that’s it, no matter how long, like 3k followers, I can simply just get that specific field</p>
